client/my-sites/sharing/connections/services-group.jsx

```diff
@@ -1,123 +1,89 @@
-			if ( 'publicize' === service.type && ! utils.userCan( 'publish_posts', site ) ) {
-				return false;
-			}
-
-			return true;
-		} );
-	},
-
-	renderService: function( service ) {
+const NUMBER_OF_PLACEHOLDERS = 4;
+
+class SharingServicesGroup extends Component {
+
+	static propTypes = {
+		site: PropTypes.object,
+		user: PropTypes.object,
+		connections: PropTypes.object,
+		onAddConnection: PropTypes.func,
+		onRemoveConnection: PropTypes.func,
+		onRefreshConnection: PropTypes.func,
+		onToggleSitewideConnection: PropTypes.func,
+		initialized: PropTypes.bool,
+		services: PropTypes.array,
+		title: PropTypes.string.isRequired,
+		description: PropTypes.string
+	};
+
+	static defaultProps = {
+		onAddConnection: function() {},
+		onRemoveConnection: function() {},
+		onRefreshConnection: function() {},
+		onToggleSitewideConnection: function() {},
+		initialized: false
+	};
+
+	renderService( service ) {
 		return <Service
-			key={ service.name }
-			site={ this.props.site }
-			user={ this.props.user }
+			key={ service.ID }
 			service={ service }
-			connections={ this.props.connections }
-			onAddConnection={ this.props.onAddConnection }
-			onRemoveConnection={ this.props.onRemoveConnection }
-			onRefreshConnection={ this.props.onRefreshConnection }
-			onToggleSitewideConnection={ this.props.onToggleSitewideConnection } />;
-	},
+			{ ...this.props } />;
+	}
 
-	renderServicePlaceholders: function() {
+	renderServicePlaceholders() {
 		// The Array constructor isn't used here because constructed arrays
 		// can't be mapped since it doesn't truly contain any values
-		return Array.apply( null, Array( NUMBER_OF_PLACEHOLDERS ) ).map( function( value, i ) {
-			return <ServicePlaceholder key={ 'service-placeholder-' + i } />;
-		} );
-	},
+		return Array.apply( null, new Array( NUMBER_OF_PLACEHOLDERS ) ).map( ( value, i ) =>
+			<ServicePlaceholder key={ 'service-placeholder-' + i } />
+		);
+	}
 
-	renderServices: function( services ) {
+	renderServices() {
 		if ( this.props.initialized ) {
-			return services.map( this.renderService, this );
+			return this.props.services.map( this.renderService, this );
 		}
 		return this.renderServicePlaceholders();
-	},
+	}
 
-	render: function() {
-		var services = this.getEligibleServices(),
-			classes = classNames( 'sharing-services-group', {
-				'is-empty': this.props.initialized && ! services.length
-			} );
+	render() {
+		const classes = classNames( 'sharing-services-group', {
+			'is-empty': this.props.initialized && ! this.props.services.length
+		} );
 
 		return (
 			<div className={ classes }>
+				<QueryKeyringServices />
 				<SectionHeader label={ this.props.title } />
 				<ul className="sharing-services-group__services">
-					{ this.renderServices( services ) }
+					{ this.renderServices() }
 				</ul>
 			</div>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state, { site, type } ) => ( {
+		services: getEligibleKeyringServices( state, site, type )
```

Comment:
We should call on `getSelectedSiteId( state )` here instead of requiring the site object be passed as a prop. This means we'd update `getEligibleKeyringServices` to accept an ID, then in its logic defer to `getSite`. Generally I'd recommend keeping props and selector arguments as simple as possible and delegating to some other selector when needing a full site object.